src/com/cloudogu/ces/cesbuildlib/K3d.groovy

```diff
@@ -0,0 +1,79 @@
+package com.cloudogu.ces.cesbuildlib
+
+import com.cloudbees.groovy.cps.NonCPS
+
+class K3d {
+    private String workspace
+    private String clusterName
+    private script
+    private String path
+
+    K3d(script, String envWorkspace, String envPath) {
+        this.workspace = envWorkspace
+        this.clusterName = createClusterName()
+        this.script = script
+        this.path = envPath
+    }
+
+    /**
+     * Creates a randomized cluster name
+     *
+     * @return new randomized cluster name
+     */
+    @NonCPS
+    static String createClusterName() {
+        String[] randomUUIDs = UUID.randomUUID().toString().split("-")
+        String uuid_snippet = randomUUIDs[randomUUIDs.length-1]
+        // Cluster name must be <= 32 characters
+        return "citest-" + uuid_snippet
+    }
+
+    /**
+     * Starts a k3d cluster in Docker
+     * Utilizes code from the cloudogu/gitops-playground
+     */
+    void startK3d() {
+        script.sh "mkdir -p ${workspace}/.k3d/bin"
```

Comment:
I didn't see this earlier when we talked about this. There are 3 occurrences of `${workspace}/.k3d/bin`. I feel this should receive its own variable (like `k3dBinaryPath` or something alike) in order to reduce the change footprint when when the directory changes or another occurrence is added in the future.